lib/using_restricted_array.rb

```diff
@@ -6,56 +6,111 @@
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the length of the array
+# Space complexity: O(1)
 def length(array)
-  raise NotImplementedError
+  length = 0
+  until array[length] == nil
+    length += 1
+  end
+  return length
 end
 
 # Prints each integer values in the array
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the length of the array
+# Space complexity: O(1)
 def print_array(array)
-  raise NotImplementedError
+  (length(array) - 1).times do |i|
+    print "#{array[i]} "
+  end
+  print array[length(array)]
```

Comment:
This will miss the last array element. 

This line should be `print array[length(array) - 1]`